### PR TITLE
Imports initial groups into temporary group store

### PR DESCRIPTION
**Pull request by tinwelint**
to later be defragmented into the main store, this to better try and
not go outside of the functionality which the PageCache provides in
order to support as many platforms as possible. This commit will
change the import behaviour regarding relationship groups

from:
- import into main group store
- defragment into replacement store
- replace main group store with the replacement store

to:
- import into temporary group store
- defragment into main store
- temporary store will be deleted on close

### CODE REVIEW

**Comment by chrisvest** [comment]
I think this looks pretty good.

**Comment by tinwelint** [comment]
Simplified some store factory utility methods in `BatchingNeoStore` and saw that it created a whole set of temporary stores, reduced it down to only the group store for now.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java [hunk 1]
```diff
@@ -220,41 +224,22 @@ public static void createStore( FileSystemAbstraction fileSystem, String storeDi
...
     private NeoStores newNeoStores()
     {
-        return newNeoStores( storeDir );
-    }
-
-    private NeoStores newNeoStores( File storeDir )
-    {
-        StoreFactory storeFactory = newStoreFactory( pageCache, recordFormats, storeDir );
-        return storeFactory.openAllNeoStores( true );
+        return newStoreFactory( DEFAULT_NAME, DELETE_ON_CLOSE ).openAllNeoStores( true );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java]: Wait, this one is for the non-temporary stores, isn't it? I don't think that should be `DELETE_ON_CLOSE`.

**Comment by chrisvest** [comment]
@tinwelint Review complete. I think I spotted a bug.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java]: Heh yeah, removing `DELETE_ON_CLOSE`. Wait... why did all the tests go green, I wonder

**Comment by tinwelint** [comment]
@chrisvest There were places where `openOptions` weren't forwarded correctly. Fixed that and added test in `StoreFactoryTest`. Now it should be OK.

**Comment by chrisvest** [comment]
Looks good. Will merge on green.

**Comment by chrisvest** [comment]
Failed on an unrelated flaky test that has been haunting us recently.

**Pull request merged**